esmvaltool/diag_scripts/lifetime/lifetime_base.py

```diff
@@ -0,0 +1,875 @@
+"""Base class for lifetime diagnostics."""
+
+import logging
+import os
+import re
+
+from copy import deepcopy
+# import cartopy
+import iris
+import numpy as np
+import matplotlib.pyplot as plt
+import yaml
+from iris.analysis import MEAN
+from iris.util import broadcast_to_shape
+from mapgenerator.plotting.timeseries import PlotSeries
+from scipy.constants import g, N_A, R
+
+from esmvaltool.diag_scripts.shared import ProvenanceLogger, names
+
+# set standard molar masses
+M_AIR = 28.970  # [g_air/mol_air]
+M_H2O = 18.02   # [g_air/mol_air]
+
+logger = logging.getLogger(__name__)
+
+
+def create_press(var):
+    """Create a pressure variable."""
+    resolver = iris.common.resolve.Resolve(var, var)
+    if var.coord('air_pressure').points.shape == var.shape:
+        press = resolver.cube(var.coord('air_pressure').points)
+    else:
+        press = resolver.cube(broadcast_to_shape(
+            var.coord('air_pressure').points,
+            var.shape,
+            var.coord_dims('air_pressure')
+        ))
+    press.long_name = 'air_pressure'
+    press.var_name = 'air_pressure'
+    press.units = 'Pa'
+
+    return press
+
+
+def calculate_gridmassdry(press, hus, z_coord):
+    """Calculate the dry gridmass according to pressure and humidity.
+
+    Used constants:
+    g  standard acceleration of gravity from scipy
+    """
+    # calculate minimum in cubes
+    if isinstance(press, iris.cube.Cube):
+        pmin = np.nanmin(press.data)
+    else:
+        pmin = np.nanmin(press)
+
+    # surface air pressure as cube
+    pmax = press[:, 0, :, :].copy()
+    pmax.data = press.coord('surface_air_pressure').points
+
+    # delta pressure > kg m-1 s-2
+    delta_p = dpres_plevel_4d(press,
+                              pmin,
+                              pmax,
+                              z_coord=z_coord)
+    # grid mass per square meter > kg m-2
+    delta_m = delta_p / g
+    # grid area > m2
+    area = calculate_area(hus.coord('latitude'),
+                          hus.coord('longitude'))
```

Comment:
You can probably replace this with

```suggestion
    lat_lon_dims = sorted(
        tuple(set(hus.coord_dims('latitude') + hus.coord_dims('longitude')))
    )
    lat_lon_slice = next(hus.slices(['latitude', 'longitude'], ordered=False))
    area_2d = iris.analysis.cartography.area_weights(lat_lon_slice)
    area = broadcast_to_shape(
        da.array(area_2d),
        hus.shape,
        lat_lon_dims,
        chunks=hus.lazy_data().chunksize,
    )
```

And remove the custom `calculate_area` code. Note that `area` is now an array, not a cube (but that shouldn't be a problem I think).

In theory you should be able to simply do `area = iris.analysis.cartography.area_weights(hus)`, but that's not fully lazy yet and will return a numpy array instead of a dask array.